chainstate/test-framework/src/failing_storage/read.rs

```diff
@@ -0,0 +1,269 @@
+// Copyright (c) 2024 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::BTreeMap;
+
+use chainstate_storage::{BlockchainStorageRead, ChainstateStorageVersion};
+use chainstate_types::{storage_result, BlockIndex, EpochData, EpochStorageRead};
+use common::{
+    chain::{
+        block::signed_block_header::SignedBlockHeader,
+        config::{EpochIndex, MagicBytes},
+        tokens::TokenId,
+        Block, DelegationId, GenBlock, PoolId, Transaction, UtxoOutPoint,
+    },
+    primitives::{Amount, BlockHeight, Id},
+};
+use pos_accounting::PoSAccountingStorageRead;
+use tokens_accounting::{TokenData, TokensAccountingStorageRead};
+use utxo::UtxosStorageRead;
+
+use super::{FailingStorage, FailingStorageTxRw};
+
+impl<T: BlockchainStorageRead> BlockchainStorageRead for FailingStorageTxRw<'_, T> {
```

Comment:
Maybe I'm missing something, but shouldn't BlockchainStorageRead be already implemented for all types of internal storage? Can't we use this somehow?

```
impl<'st, B: storage::Backend> BlockchainStorageRead for super::StoreTxRo<'st, B>
```